### PR TITLE
Two new problems

### PR DESCRIPTION
**Pull request by iamsavva**
Two new problems:
one on Lyapunov Instability / Barrier functions
one on Converse Lyapunov Theorem

- - -
This change is [Reviewable](https://reviewable.io/reviews/RussTedrake/underactuated/541)

### CODE REVIEW

**Review by jaehunchoe827 — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RussTedrake/underactuated/541)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @iamsavva)

___
*[`book/lyapunov.html` line 2141 at r1](https://reviewable.io/reviews/RussTedrake/underactuated/541#-NrlSQCj84RfwPVUmZ8u:-NrlSQCj84RfwPVUmZ8v:bxvhem2) ([raw file](https://github.com/RussTedrake/underactuated/blob/d130d9466462fac1c6e49c8c78bb359c87135986/book/lyapunov.html#L2141)):*
> ```HTML
>             Suppose our system starts at $\bx(0)$ where 
>             $V\big(\bx(0)\big) < 0$. Can there be a time $t$, such that  
>             $V\big(\bx(t)\big) \geq 0$? 
> ```

We've assumed that V(x) <= 0 for all x.
Wouldn't this assumption make this question be identical to the first question?
I guess you want to ask some general question here?
___
*[`book/lyapunov.html` line 2199 at r1](https://reviewable.io/reviews/RussTedrake/underactuated/541#-NrlUsQr88UlLa-bBFaz:-NrlUsQr88UlLa-bBFb-:b-4os31h) ([raw file](https://github.com/RussTedrake/underactuated/blob/d130d9466462fac1c6e49c8c78bb359c87135986/book/lyapunov.html#L2199)):*
> ```HTML
>           <li>
>             Suppose you have a dynamical system $\dot \bx = f(\bx)$. 
>             Why can't you "simply" compute $V(\mathbf z)$ as above to check 
> ```

I don't quite understand this; does it mean to compute V(z) and express it as a function of z in closed form?

**Review by iamsavva — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RussTedrake/underactuated/541)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @jaehunchoe827)

___
*[`book/lyapunov.html` line 2141 at r1](https://reviewable.io/reviews/RussTedrake/underactuated/541#-NrlSQCj84RfwPVUmZ8u:-NrlYyVuDIkfInaaY-kD:b-nd9w05) ([raw file](https://github.com/RussTedrake/underactuated/blob/d130d9466462fac1c6e49c8c78bb359c87135986/book/lyapunov.html#L2141)):*
<details><summary><i>Previously, jaehunchoe827 wrote…</i></summary><blockquote>

We've assumed that V(x) <= 0 for all x.
Wouldn't this assumption make this question be identical to the first question?
I guess you want to ask some general question here?
</blockquote></details>

We did not assume that V(X) <= 0 for all x. Rather, we assumed that for all x, such that V(x) <=0, \dot V(x) <= 0.

indeed, part d is similar to a. But what we are asking is different: 
In (a) we ask if the system converges to 0 as t goes to infinity -- it doesn't -- hence it's not GAS.
In (d), we ask if the system can EVER be in a state that has nonnegative V -- it can't -- hence V can be interpreted as a barrier to the set {x | V(x) >= 0}. I'll rephrase it to make the later point more clear -- thanks.
___
*[`book/lyapunov.html` line 2199 at r1](https://reviewable.io/reviews/RussTedrake/underactuated/541#-NrlUsQr88UlLa-bBFaz:-Nrl_Hmo3grgz4jwGR2b:bdijq8u) ([raw file](https://github.com/RussTedrake/underactuated/blob/d130d9466462fac1c6e49c8c78bb359c87135986/book/lyapunov.html#L2199)):*
<details><summary><i>Previously, jaehunchoe827 wrote…</i></summary><blockquote>

I don't quite understand this; does it mean to compute V(z) and express it as a function of z in closed form?
</blockquote></details>

yes. compute can mean anything -- in closed analytic form, or numerically, doesn't matter.
the point we are trying to make here is that although V(z) always exists for any GAS system, V(z) is actually really hard to compute, since integration is hard. hence although at first sight it may seem like checking for GAS is easy -- it's actually not.

**Review by jaehunchoe827 — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/RussTedrake/underactuated/541#-:-Nrlf4q1DHuUU5ENpU17:bnfp4nl)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @iamsavva)

___
*[`book/lyapunov.html` line 2141 at r1](https://reviewable.io/reviews/RussTedrake/underactuated/541#-NrlSQCj84RfwPVUmZ8u:-Nrle8lNE6PsaayQoupM:bbo3kjd) ([raw file](https://github.com/RussTedrake/underactuated/blob/d130d9466462fac1c6e49c8c78bb359c87135986/book/lyapunov.html#L2141)):*
<details><summary><i>Previously, iamsavva (Savva Morozov) wrote…</i></summary><blockquote>

We did not assume that V(X) <= 0 for all x. Rather, we assumed that for all x, such that V(x) <=0, \dot V(x) <= 0.

indeed, part d is similar to a. But what we are asking is different: 
In (a) we ask if the system converges to 0 as t goes to infinity -- it doesn't -- hence it's not GAS.
In (d), we ask if the system can EVER be in a state that has nonnegative V -- it can't -- hence V can be interpreted as a barrier to the set {x | V(x) >= 0}. I'll rephrase it to make the later point more clear -- thanks.
</blockquote></details>

Oops, my bad, I misinterpreted that. It all makes sense, thanks!
___
*[`book/lyapunov.html` line 2199 at r1](https://reviewable.io/reviews/RussTedrake/underactuated/541#-NrlUsQr88UlLa-bBFaz:-NrlebMk-hCCnHKIAB03:b297hwn) ([raw file](https://github.com/RussTedrake/underactuated/blob/d130d9466462fac1c6e49c8c78bb359c87135986/book/lyapunov.html#L2199)):*
<details><summary><i>Previously, iamsavva (Savva Morozov) wrote…</i></summary><blockquote>

yes. compute can mean anything -- in closed analytic form, or numerically, doesn't matter.
the point we are trying to make here is that although V(z) always exists for any GAS system, V(z) is actually really hard to compute, since integration is hard. hence although at first sight it may seem like checking for GAS is easy -- it's actually not.
</blockquote></details>

I see your point, thanks for the clarification.

**Review by iamsavva — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RussTedrake/underactuated/541)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @jaehunchoe827)

___
*[`book/lyapunov.html` line 2141 at r1](https://reviewable.io/reviews/RussTedrake/underactuated/541#-NrlSQCj84RfwPVUmZ8u:-NrmTjlN1of_22y-Vrct:b-896fix) ([raw file](https://github.com/RussTedrake/underactuated/blob/d130d9466462fac1c6e49c8c78bb359c87135986/book/lyapunov.html#L2141)):*
<details><summary><i>Previously, jaehunchoe827 wrote…</i></summary><blockquote>

Oops, my bad, I misinterpreted that. It all makes sense, thanks!
</blockquote></details>

Done.
___
*[`book/lyapunov.html` line 2199 at r1](https://reviewable.io/reviews/RussTedrake/underactuated/541#-NrlUsQr88UlLa-bBFaz:-NrmTiXj3LQVwMJUeHVx:b-896fix) ([raw file](https://github.com/RussTedrake/underactuated/blob/d130d9466462fac1c6e49c8c78bb359c87135986/book/lyapunov.html#L2199)):*
<details><summary><i>Previously, jaehunchoe827 wrote…</i></summary><blockquote>

I see your point, thanks for the clarification.
</blockquote></details>

Done.

**Review by RussTedrake — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RussTedrake/underactuated/541)* status: all files reviewed, 2 unresolved discussions (waiting on @jaehunchoe827)
